go-runtime/ftl/ftltest/ftltest.go

```diff
@@ -43,31 +37,55 @@ func Context(options ...func(context.Context) error) context.Context {
 
 // WithConfig sets a configuration for the current module
 //
-// To be used with Context(...)
-func WithConfig(name string, value any) func(context.Context) error {
+// To be used when setting up a context for a test:
+// ctx := ftltest.Context(
+//
+//	ftltest.WithConfig(exampleEndpoint, "https://example.com"),
+//	... other options
+//
+// )
+func WithConfig[T ftl.ConfigType](config ftl.ConfigValue[T], value T) func(context.Context) error {
 	return func(ctx context.Context) error {
+		if config.Module != ftl.Module() {
+			return fmt.Errorf("config %v does not match current module %s", config.Module, ftl.Module())
+		}
 		cm := cf.ConfigFromContext(ctx)
-		return cm.Set(ctx, cf.Ref{Module: optional.Some(ftl.Module()), Name: name}, value)
+		return cm.Set(ctx, cf.Ref{Module: optional.Some(config.Module), Name: config.Name}, value)
 	}
 }
 
 // WithSecret sets a secret for the current module
 //
-// To be used with Context(...)
-func WithSecret(name string, value any) func(context.Context) error {
+// To be used when setting up a context for a test:
+// ctx := ftltest.Context(
+//
+//	ftltest.WithSecret(privateKey, "abc123"),
+//	... other options
+//
+// )
+func WithSecret[T ftl.SecretType](secret ftl.SecretValue[T], value T) func(context.Context) error {
 	return func(ctx context.Context) error {
-		cm := cf.SecretsFromContext(ctx)
-		return cm.Set(ctx, cf.Ref{Module: optional.Some(ftl.Module()), Name: name}, value)
+		if secret.Module != ftl.Module() {
+			return fmt.Errorf("secret %v does not match current module %s", secret.Module, ftl.Module())
+		}
+		sm := cf.SecretsFromContext(ctx)
```

Comment:
BTW I had been meaning to mention this, but the configuration code isn't intended for use in the go-runtime. I'll file a ticket to get that removed.